nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -288,10 +293,18 @@ geometry_msgs::msg::TwistStamped RegulatedPurePursuitController::computeVelocity
     (carrot_pose.pose.position.x * carrot_pose.pose.position.x) +
     (carrot_pose.pose.position.y * carrot_pose.pose.position.y);
 
+  // similarly find dist^2 to half of look_ahead_point to smooth the curvature
+  auto mid_pose = getLookAheadPoint(lookahead_dist / 2.0, transformed_plan);
```

Comment:
As I mentioned in Slack, please revert all of the mid_pose stuff, that's probably going to break the stability analysis of the algorithm and you can accomplish a very similar (or exactly the same?) thing just by lowering the lookahead distance to begin with. This is trying to fix an unrelated issue to the collision checking on approach to the goal and using the full SE2 footprint. I think you've mentioned this happens regardless of these changes (eg you see this behavior even when using just point-costs), so its an unrelated topic that is probably just from poor defaults for this particular environment.

Reply:
If you lower the lookahead distances, doesn't that just fix the problem as well? Keep in mind these are just defaults I set in testing in maps while developing the algorithm, there's nothing special about them.